src/linalg.jl

```diff
@@ -149,3 +149,21 @@ function _mul!(C::BlockDiagonal, A::BlockDiagonal, B::BlockDiagonal, α::Number,
 
     return C
 end
+
+# There is no restriction on x to be able to pass Cholesky objexts
+function issquare(x)
+    indsm, indsn = size(x)
+    n1 == n2
+end
+
+function LinearAlgebra.:\(B::BlockDiagonal, vm::AbstractVecOrMat)
+    row_i = 1
+    all(issquare, blocks(B)) || error("left division is not compatible with non-square matrices")
+    result = similar(vm)
+    for block in blocks(B)
+        nrow = size(block, 1)
+        result[row_i:row_i+nrow-1, :] = block \ vm[row_i:row_i+nrow-1, :]
+        row_i += nrow
+    end
+    return result
+end
```

Comment:
Interestingly:
- left division is _in principle_ defined for square matrices that are built up from non-square blocks, e.g. `b = BlockDiagonal([rand(1, 2), rand(2, 1)])`
- however, I think that the determinant of these matrices is always zero (and therefore the left division is not defined)

so maybe something like:

```suggestion
function LinearAlgebra.:\(B::BlockDiagonal, vm::AbstractVecOrMat)
    row_i = 1
    # BlockDiagonals with non-square blocks
    all(b -> size(b, 1) == size(b, 2), blocks(B)) || throw(
        ArgumentError(
            "Left division (\\) is not defined for BlockDiagonals with " *
            "non-square blocks (they are singular matrices).",
        ),
    )
    result = similar(vm)
    for block in blocks(B)
        nrow = size(block, 1)
        result[row_i:(row_i + nrow - 1), :] = block \ vm[row_i:(row_i + nrow - 1), :]
        row_i += nrow
    end
    return result
end
```